### ISSUE TITLE
the input/output of super resolution are different

### ISSUE THREAD

**Issue by guoyejun** [opencv/open_model_zoo#1109]
i tried the code at https://github.com/opencv/open_model_zoo/tree/master/demos/super_resolution_demo with three models:  single-image-super-resolution-1032, single-image-super-resolution-1033 and text-image-super-resolution-0001.

The input and output of the three models are all FP32, but, the input value range is [0.0, 250.0], while the output value is normalized (with range [0.0, 1.*]). And application have to clamp it to [0.0, 1.0].

Just think it would be better that if the input/output range are both [0.0, 1.0], and not letting application do extra clamp since image's valid range [0.0, 1.0]

**Comment by vladimir-dudnik** [comment]
@snosov1 could you please comment on this?

**Comment by snosov1** [comment]
I guess, from the "clarity of API" perspective, it makes sense. But it kinda moves the model further away from its original (i.e. training framework) counter-part. There's a lot of things one can push into the model beyond clamps - things, like, resize or color conversion, too. 

But where do you draw the line? Say, the user gets [0..255] values naturally. If we accept [0..1] range, then the user has to do not only the FP32 conversion, but scaling as well - this is extra cycles and an extra step at the app level (that you want to avoid).

Accepting RGB can also be challenged. If the user works with encoded video streams - he'll get the frames in NV12 format most of the time. So, ideally, the model should accept NV12 directly. 

What I'm saying is that there's many different combinations of input and output formats and the "correct" one depends on the end application. The right way to do this is to handle it during training and export for inference accordingly. For us, it's unfeasible to maintain all the different combinations, so we just leave the model as close to the training set up as possible and make all the fiddling at the app level.

**Comment by guoyejun** [comment]
thanks. and yes, i look at it from the "clarity of API" perspective. The background is that I want to support this model in FFmpeg's filter dnn_processing which targets to be a general filter to load all kinds of DL models with 'unified API'.

**Comment by snosov1** [comment]
I see. But I'm afraid you won't get away without some kind of app-level "adapters" anyway. Even if we make the SR networks comply to your _current_ API reqs - there will still be networks with other input/output conditions or your _later_ API reqs may change. 

One option to do this is to embed the necessary layers into the IR semi-automatic (i.e. you load the IR into Python and tweak it - add necessary layers, merge inputs, etc.). This has to be done by someone in any case - either as a post-training/pre-export step or at the MO conversion stage.

Our training code for super resolution is here: https://github.com/opencv/openvino_training_extensions/tree/develop/pytorch_toolkit/super_resolution You can either use it as a reference or submit a patch to the `export.py` script that will do the necessary steps conditionally.

**Comment by guoyejun** [comment]
thanks, and another question, why we fix the input dims of the model as 360x640. I saw some other models with width=-1 and height=-1, the model will be more general to accept all kinds of image sizes.

**Comment by snosov1** [comment]
There're some optimizations that happen on Model Optimizer step that will be unavailable if we convert the model to support various input resolutions. So, currently, if one wants to change the resolution - it should be done on MO step.

That said, I think, we can accept the contribution of a "dynamic model" if you care to submit. You can find the original model at https://github.com/opencv/openvino_training_extensions/tree/develop/pytorch_toolkit/super_resolution
